internal/impl/babysitter.go

```diff
@@ -193,8 +218,8 @@ func (b *babysitter) watchPods(ctx context.Context, component string) error {
 			return ctx.Err()
 
 		case event, ok := <-watcher.ResultChan():
-			if !ok {
-				return nil
+			if !ok { // the watcher should attempt to handle errors on the channel
```

Comment:
If `ok` is false, then `watcher.ResultChan` is closed, so continuing in this loop will cause an infinite loop until the context is cancelled.